### PR TITLE
ENH lint for no `.ci_support` files and thus no package builds

### PR DESCRIPTION
**Pull request by beckermr**
Checklist
* [x] Added a ``news`` entry

xref: https://github.com/regro/cf-scripts/issues/2627

### CODE REVIEW

File: tests/test_lint_recipe.py [hunk 1]
```diff
@@ -1772,5 +1772,25 @@ def assert_jinja(jinja_var, is_good=True):
             assert_jinja('{% set version= "0.27.3"%}', is_good=False)
 
 
+def test_lint_no_builds():
+    expected_message = "The feedstock has no `.ci_support` files and "
+
+    with tmp_directory() as feedstock_dir:
+        ci_support_dir = os.path.join(feedstock_dir, "recipe")
```
Comment (beckermr) [inline tests/test_lint_recipe.py:1779]:
```suggestion
        ci_support_dir = os.path.join(feedstock_dir, ".ci_support")
```

**Comment by beckermr** [comment]
@conda-forge/core This PR is ready for review!

File: conda_smithy/lint_recipe.py [hunk 2]
```diff
@@ -1191,6 +1191,17 @@ def run_conda_forge_specific(meta, recipe_dir, lints, hints):
...
 
+    # 7: Ensure that the recipe has some .ci_support files
+    if not is_staged_recipes and recipe_dir is not None:
+        ci_support_files = glob(
+            os.path.join(recipe_dir, "..", ".ci_support", "*.yaml")
+        )
+        if not ci_support_files:
+            lints.append(
+                "The feedstock has no `.ci_support` files and thus will not build any packages. "
+                "Please check that this is the intended behavior."
```
Comment (h-vetinari) [inline conda_smithy/lint_recipe.py:1202]: Can this ever be the intended behaviour? I don't think that's possible/reasonable? Perhaps easier to just delete this line...
```suggestion
```

**Comment by beckermr** [comment]
Done! Shall we merge @h-vetinari!

File: conda_smithy/lint_recipe.py [hunk 3]
```diff
@@ -1191,6 +1191,16 @@ def run_conda_forge_specific(meta, recipe_dir, lints, hints):
...
             )
 
+    # 7: Ensure that the recipe has some .ci_support files
+    if not is_staged_recipes and recipe_dir is not None:
+        ci_support_files = glob(
+            os.path.join(recipe_dir, "..", ".ci_support", "*.yaml")
+        )
+        if not ci_support_files:
+            lints.append(
+                "The feedstock has no `.ci_support` files and thus will not build any packages. "
```
Comment (beckermr) [inline conda_smithy/lint_recipe.py:1201]:
```suggestion
                "The feedstock has no `.ci_support` files and thus will not build any packages."
```

**Review by h-vetinari — APPROVED** [review]
👍

**Comment by jakirkham** [comment]
Thanks Matt! 🙏

This is a great idea. Even outside the case you identified, we also periodically have maintainers come to us say their feedstock didn't upload anything when they effectively skipped all builds. Hopefully this will help flag those cases before they merge such PRs

Wonder if there is a way to catch this sort of issue in `staged-recipes` too

**Pull request merged**